compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -291,26 +345,25 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             | ty::InstanceDef::CloneShim(..)
             | ty::InstanceDef::Item(_) => {
                 // We need MIR for this fn
-                let body =
+                let (body, instance) =
                     match M::find_mir_or_eval_fn(self, instance, caller_abi, args, ret, unwind)? {
                         Some(body) => body,
                         None => return Ok(()),
                     };
 
-                // Check against the ABI of the MIR body we are calling (not the ABI of `instance`;
-                // these can differ when `find_mir_or_eval_fn` does something clever like resolve
-                // exported symbol names).
-                let callee_def_id = body.source.def_id();
-                let callee_abi = get_abi(self, self.tcx.type_of(callee_def_id));
+                // Compute callee information using the `instance` returned by
+                // `find_mir_or_eval_fn`.
+                let callee_abi = get_abi(self, instance.ty(*self.tcx, self.param_env));
+                // FIXME: for variadic support, do we have to somehow determine calle's extra_args?
```

Comment:
You would use the exact same `extra_args` types for both caller and callee, as determined from the callsite. If you just want to pass the `caller_fn_abi` to this point, as the sole source of truth, you should be able to recover the types from `caller_fn_abi.args[caller_fn_abi.fixed_count..].iter().map(|arg| arg.layout.ty)`.

Reply:
Uh okay that's weird, how would the callee "know" about that in a regular codegen backend where it doesn't have access to the call site? Just passing the extra_args here will lead to an ICE later since then there are more arguments in the FnAbi than the MIR body expects.

But I wasn't planning on implementing variadics in this PR (or any time soon, really), so I think we can just leave this aside for now.